ethcore/src/client/client.rs

```diff
@@ -893,14 +893,13 @@ impl BlockChainClient for Client {
 			ExecutionError::TransactionMalformed(message)
 		})?;
 		let balance = original_state.balance(&sender);
-		let needed_balance = t.value + t.gas * t.gas_price;
+		let needed_balance = t.value + U256::from(UPPER_CEILING) * t.gas_price;
```

Comment:
there's already a lot of these in the codebase (e.g. `Client::call`) - better to sweep them all at once if it's actually required.